File: zb1/codingTest/220317/NthCanceledOrder/NthCanceledOrder.js
/**
 * @param orders {number[]}
 * @param n {number}
 * @returns {number}
 */
function solution(orders, n) {
  let missCount = 0
  let num = 0
  let i = 0
  while (missCount !== n) {
    num++

    if (orders[i] === num) {
      i++
    } else {
      missCount++
    }
  }
  return num
}

solution

File: zb1/codingTest/220317/NthCanceledOrder/NthCanceledOrder.py
def solution(orders, n):
    """
    :param orders: int[]
    :param n: int
    :return: int
    """

    missCount = 0
    num = 0
    i = 0
    while missCount != n:
        num += 1

        if i < len(orders) and orders[i] == num:
            i += 1
        else:
            missCount += 1
    return num
